src/main/java/org/snomed/snowstorm/rest/CodeSystemController.java

```diff
@@ -178,9 +181,21 @@ public CodeSystemVersion updateVersion(@PathVariable String shortName, @PathVari
 					"to support creating a new version of the extension. \n\n" +
 					"If you are the extension maintainer an integrity check should be run after this operation to find content that needs fixing. ")
 	@PostMapping(value = "/{shortName}/upgrade")
-	public void upgradeCodeSystem(@PathVariable String shortName, @RequestBody CodeSystemUpgradeRequest request) throws ServiceException {
+	public ResponseEntity<Void> upgradeCodeSystem(@PathVariable String shortName, @RequestBody CodeSystemUpgradeRequest request) throws ServiceException {
```

Comment:
Good use of ResponseEntity to deal with both sync and async styles of call.

Reply:
As discussed, I note that we're removing the option to use this method in a synchronous manner.   I think that does simplify the problem greatly.